bokeh/driving.py

```diff
@@ -162,11 +162,11 @@ def repeat(sequence):
 
     '''
     N = len(sequence)
-    def f(i):
+    def f(i: int) -> int:
         return sequence[i%N]
     return partial(force, sequence=_advance(f))
 
-def sine(w, A=1, phi=0, offset=0):
+def sine(w: float, A: float =1, phi: float =0, offset: float =0) -> Callable:
```

Comment:
I actually don't love ` =x` with a leading space, I'd really just prefer everything close in:

```
def sine(w: float, A:float=1, phi: float=0, offset: float=0) -> Callable
```
@mattpap can we agree on a format?

Reply:
The standard in the typing community is `arg: Type = default`. Otherwise things get unreadable pretty quickly, and it's a better default when spreading arguments over multiple lines, which is pretty common when including type annotations in large function signatures. Personally I never liked `arg=default` to begin with. [Here](https://github.com/tornadoweb/tornado/blob/master/tornado/escape.py#L275-L281) is an example from Tornado, which btw. is fully statically typed.

Reply:
OK that will take some getting used to but is preferable to space/no-space combination